src/Adafruit_TinyUSB.h

```diff
@@ -36,11 +36,19 @@
 #if TUSB_OPT_DEVICE_ENABLED
 
 #include "arduino/Adafruit_USBD_Device.h"
-#include "arduino/Adafruit_USBD_CDC.h"
+#if CFG_TUD_CDC
+  #include "arduino/Adafruit_USBD_CDC.h"
+#endif
 
-#include "arduino/hid/Adafruit_USBD_HID.h"
-#include "arduino/midi/Adafruit_USBD_MIDI.h"
-#include "arduino/msc/Adafruit_USBD_MSC.h"
+#if CFG_TUD_HID
+  #include "arduino/hid/Adafruit_USBD_HID.h"
+#endif
+#if CFG_TUD_MIDI
+  #include "arduino/midi/Adafruit_USBD_MIDI.h"
+#endif
+#if CFG_TUD_MSC
+  #include "arduino/msc/Adafruit_USBD_MSC.h"
+#endif
```

Comment:
do you have any particular issues with these include when its class driver are not enabled ?

Reply:
I had a custom build with 

#define CFG_TUD_HID 0

it was causing a build failure with 'undefined type: hid_report_type_t' message.

The patch above had fixed the build.

Reply:
can you provide more info to reproduce:
- your board
- bsp core & library version
- which sketch you are compiling

Reply:
Sure.

* PCA10056 build target
* Core 1.1.0 with Adafruit_TinyUSB_Arduino 1.4.4
* Adafruit_TinyUSB_Arduino/examples/CDC/cdc_multi

HID component is disabled in Adafruit_TinyUSB_Arduino/src/arduino/ports/nrf/tusb_config_nrf.h as follows:

//------------- CLASS -------------//
#define CFG_TUD_CDC 1
#define CFG_TUD_MSC 1
#define CFG_TUD_HID 0
#define CFG_TUD_MIDI 1
#define CFG_TUD_VENDOR 1

I've got a build error message:

In file included from /home/user/Arduino/hardware/adafruit/nrf52/libraries/Adafruit_TinyUSB_Arduino/src/Adafruit_TinyUSB.h:44,
                 from /home/user/Arduino/hardware/adafruit/nrf52/libraries/Adafruit_TinyUSB_Arduino/examples/CDC/cdc_multi/cdc_multi.ino:20:
/home/user/Arduino/hardware/adafruit/nrf52/libraries/Adafruit_TinyUSB_Arduino/src/arduino/hid/Adafruit_USBD_HID.h:33:45: error: 'hid_report_type_t' has not been declared
   33 |                                             hid_report_type_t report_type,